DOC: add documentation on why astropy has tables (#17520)
astropy/astropy: docs/table/table_and_dataframes.rst

metadata to allow tracking of the units of the data for applications not using
|Quantity|.

Pandas does not provide support for units.

Multi-dimensional and Structured Columns
----------------------------------------

Astronomers deal with images, spectra, and other multi-dimensional data that are
commonly stored in a table. An example is a source catalog with an image thumbnail and a
spectrum for each source. Structured columns are less common, but are useful for storing
vectorized data like an `~astropy.coordinates.EarthLocation` in a table.

Pandas is not able to natively store multi-dimensional or structured columns.

Lossless representation of FITS and VOTable data via metadata
-------------------------------------------------------------

The `astropy.table` package is designed to provide lossless representation of FITS and
VOTable data. This means that when you read a FITS or VOTable file into a table and then
write it back out, the data will be effectively identical. This is made possible by
robust support for table and column metadata which allows storing and propagating common
column information such as the unit, description, and format. For VOTable data, more

> esoteric information like the UCD is maintained.

Reviewer: can we use another, more neutral adjective here? Or even just drop the adjective
```suggestion information like the UCD is maintained.
```

Author: Fixed.

> FITS and VOTable are two important file formats in astronomy, but the idea of lossless

Reviewer: This is absolute nitpicking, but my understanding is that VOTable is a format standard and not a file format standard, and serialization in a format other than XML is theoretically possible. 

Less nitpicky, is that I don't really understand the point this paragraph tries to make in regard of astroquery

Author: Fixed both, hopefully.

"experimental" in the documentation.

Time and Coordinates
--------------------

Time and coordinates are fundamental to astronomy, and astropy provides robust support
for them with the `~astropy.time.Time` and `~astropy.coordinates.SkyCoord` classes.
Arrays of times and coordinates can be natively stored in `astropy.table`, meaning that
the full power of these objects is available when working with them as columns within a
table.

Pandas supports `timeseries
<https://pandas.pydata.org/docs/user_guide/timeseries.html>`_ data, but with key
limitations:

- Leap seconds are not supported. In many circumstances (for instance planning an
  observation) this limitation is not acceptable.
- Pandas times are stored with 64-bit precision, which is not sufficient for some
  astronomical applications. Astropy uses 128-bit precision for time to allow
  sub-nanosecond precision over the age of the universe.
- Different :ref:`time scales <time-scale>` common in astronomy (e.g. TAI, UT1) are
  not supported.
- :ref:`Time formats <time-format>` used in astronomy such as the FITS time format are
  not supported.

> Pandas does not support coordinate columns.

Reviewer: this is significant, maybe you want to divide this section into two, one for time and one for coordinates?

and analysis.

Units and Quantities
--------------------

Astronomy is a physical science, and the data often have units associated with
them. The `astropy.table` package natively supports |Quantity| columns, which are a
powerful way to attach units to array data and perform unit-aware operations. In
addition, the base `~astropy.table.Column` class holds a ``unit`` attribute as
metadata to allow tracking of the units of the data for applications not using
|Quantity|.

*Pandas does not provide support for units.*

Multi-dimensional and Structured Columns
----------------------------------------

Astronomers deal with images, spectra, and other multi-dimensional data that are
commonly stored in a table. An example is a source catalog with an image thumbnail and a
spectrum for each source. Structured columns are less common, but are useful for storing
vectorized data like an `~astropy.coordinates.EarthLocation` in a table.

*Pandas is not able to natively store multi-dimensional or structured columns.*

Lossless representation of FITS and VOTable data via metadata
-------------------------------------------------------------

> The `astropy.table` package is designed to provide lossless representation of FITS and

Author: Hence the "is designed" qualifier. We could go to something one level lower like "strives to".

> The `astropy.table` package is designed to provide lossless representation of FITS and

Reviewer 2: FWIW I would prefer "strives to" because I cannot tell from the text if this "design" is fulfilled or not.

> The `astropy.table` package is designed to provide lossless representation of FITS and

Author: Can you make a suggestion. I'm committing all the rest of yours.

> The `astropy.table` package is designed to provide lossless representation of FITS and

Reviewer 2: ```suggestion
The `astropy.table` package strives to provide lossless representation of FITS and
```

> The `astropy.table` package is designed to provide lossless representation of FITS and

Reviewer 2: Suggestion provided in https://github.com/astropy/astropy/pull/17520/files#diff-4bd7af647ce2840ff6686f1fbccf9e767f22776f4e647cef7765df8066f8cce6 . Thanks!

"experimental" in the documentation.*

Time and Coordinates
--------------------

Time and coordinates are fundamental to astronomy, and astropy provides robust support
for them with the `~astropy.time.Time` and `~astropy.coordinates.SkyCoord` classes.
Arrays of times and coordinates can be natively stored in `astropy.table`, meaning that
the full power of these objects is available when working with them as columns within a
table.

*Pandas supports* `timeseries
<https://pandas.pydata.org/docs/user_guide/timeseries.html>`_ *data, but with key
limitations*:

- Leap seconds are not supported. In many circumstances (for instance planning an
  observation) this limitation is not acceptable.
- Pandas times are stored with 64-bit precision, which is not sufficient for some
  astronomical applications. Astropy uses 128-bit precision for time to allow
  sub-nanosecond precision over the age of the universe.
- Different :ref:`time scales <time-scale>` common in astronomy (e.g. TAI, UT1) are
  not supported.
- :ref:`Time formats <time-format>` used in astronomy such as the FITS time format are
  not supported.

> *Pandas does not support coordinate columns.*

Reviewer 2: ```suggestion
*Pandas does not support sky coordinate columns.*
```

Responsiveness to Community Needs
---------------------------------

The `astropy.table` package is developed by the Astropy community, which is focused on
the needs of astronomers and astrophysicists. This means that the development of the
package can be responsive to the needs of this community and we can develop features
without being constrained by the potential impact to the far broader user base of
Pandas.

Interoperability
----------------

We recognize that Pandas is a popular library and that there are many users who are
familiar with it. For this reason, we have made it easy to convert between
`astropy.table` and `~pandas.DataFrame`, as documented in :ref:`pandas`. This allows

> users to take advantage of the features of both packages as needed.

Reviewer 2: ```suggestion
users to take advantage of the features of both packages as needed,
within the limitations stated above.
```

write it back out, the data will be effectively identical. This is made possible by
robust support for table and column metadata which allows storing and propagating common
column information such as the unit, description, and format. For VOTable data, more
information like the UCD is maintained.

FITS and VOTable are two important data formats in astronomy, but the idea of lossless
representation of physical data is critical in other contexts as well. For
example, accessing metadata like units is a key component of using the
`astroquery <https://astroquery.readthedocs.io/en/latest/>`_ service for astronomical

> data analysis.

Reviewer: I still don't get the astroquery reference, and why this is a key component for astroquery. We could return whatever comes back from the server, typically a VOTable or csv or whatever text file object. It's just a compatibility thing that we went a step further and parse the results into an astropy Table. 

(and returning pandas definitely brought up by a couple of modules but we pushed them to do astropy Tables instead to be more consistent with the older modules).

Author: IMO a key feature of astroquery is exactly the hard work of parsing heterogenous results from data providers into a uniform and predictable data structure (QTable) that is easy for users to work with. Many of those datasets include physical quantities with units, column descriptions, or other metadata, so if astroquery returned a DataFrame then that information would be lost.

That said, if you really don't want this shout-out to astroquery then I can replace it with a different example.

Reviewer 3: Seconded - for me part of the beauty of astroquery is exactly that it produces a table with usable columns!

Reviewer: Yes, I think if you can go with a different example it would be OK as I'm just not comfortable with the 
` is a key component of using the astroquery service for astronomical data analysis` summary, and nitpicking the sentence away is not very constructive.
